src/main/java/com/microsoft/azure/spring/data/cosmosdb/core/query/DocumentQuery.java

```diff
@@ -55,7 +55,12 @@ private boolean isCrossPartitionQuery(@NonNull String keyName) {
 
         final Optional<Criteria> criteria = this.getSubjectCriteria(this.criteria, keyName);
 
-        return criteria.map(criteria1 -> criteria1.getType() == CriteriaType.IS_EQUAL).orElse(true);
+        return criteria.map(criteria1 -> criteria1.getType() != CriteriaType.IS_EQUAL).orElse(true);
```

Comment:
For case isEqual and other condition, seems the result is opposite? e.g., `findByNameAndAgeGreaterThan`, will return `true`, but actually should be `false`?

Reply:
typo from last PR, should be `!=` here. Yes, the result is opposite. If not equal and OR keyword, should enable CrossPartitionKey.